lib/cadence/connection/thrift.rb

```diff
@@ -337,16 +343,56 @@ def get_search_attributes
         raise NotImplementedError
       end
 
-      def respond_query_task_completed
-        raise NotImplementedError
+      def respond_query_task_completed(namespace:, task_token:, query_result:)
+        query_result_proto = Serializer.serialize(query_result)
```

Comment:
You'll need to sync the changes from `lib/temporal/connection/serializer.rb` to `lib/cadence/workflow/serializer.rb`.

Do the same with the `query_answer.rb` and `query_failure.rb` files, you can put them in `lib/cadence/workflow/serializer` directory